### ISSUE TITLE
[React Native Gallery]: Visual looking headings,"View page code at GitHub", "Documentation", "Feedback" are not defined as heading.

### ISSUE THREAD

**Issue by YajurG** [microsoft/react-native-gallery#355]
**Repro-Steps:**
Launch the React Native Gallery app.
Navigate to any component (For e.g., Checkbox) and invoke it.
Set narrator to headings mode using caps lock + Page up/down.
Navigate in heading mode and listen to the narrator announcement.

**Actual Result:**
Visual looking headings,"View page code at GitHub", "Documentation", "Feedback" are not defined as heading. When user navigates to these headings narrator simply announcing as static text.

**Expected Result:**
Visual headings should be defined as the heading programmatically as well. When user navigates to "View page code at GitHub", "Documentation", "Feedback" narrator should announce as "View page code at GitHub, heading at level 2" and so on.

**User Impact:**
Users with visual impairments who rely on screen reader may use heading navigation mode to render through the page. Not defining heading in the application will make it difficult for users navigating in the app.

Internal Bug Link: https://dev.azure.com/microsoft/OS/_workitems/edit/45860903
